crates/created-swarm/src/swarm.rs

```diff
@@ -91,10 +93,11 @@ pub async fn make_swarms_with_cfg<F>(n: usize, mut update_cfg: F) -> Vec<Created
 where
     F: (FnMut(SwarmConfig) -> SwarmConfig),
 {
+    let network_key = NetworkKey::random();
     make_swarms_with(
         n,
         move |bs, maddr| {
-            let cfg = update_cfg(SwarmConfig::new(bs, maddr));
+            let cfg = update_cfg(SwarmConfig::new(bs, maddr, network_key.clone()));
```

Comment:
Why not to inline? Like
```suggestion
            let cfg = update_cfg(SwarmConfig::new(bs, maddr, NetworkKey::random()));
```